packages/nuxt3/src/core/plugins/unctx.ts

```diff
@@ -0,0 +1,31 @@
+import { Nuxt, NuxtApp, NuxtMiddleware } from '@nuxt/schema'
+import { createTransformer } from 'unctx/transform'
+import { createUnplugin } from 'unplugin'
+
+export const UnctxTransformPlugin = (nuxt: Nuxt) => {
+  const transformer = createTransformer({
+    asyncFunctions: ['defineNuxtPlugin', 'defineNuxtRouteMiddleware', 'withAsyncContext', 'callAsync']
```

Comment:
unctx, concats and preserves built-ins (https://github.com/unjs/unctx/blob/main/src/transform.ts#L31)

```suggestion
    asyncFunctions: ['defineNuxtPlugin', 'defineNuxtRouteMiddleware']
```

Reply:
Interesting, but I actually hesitate about this behavior of `unctx`, since then you can't disable the transformation of those built-in entries.

Reply:
Well, to be fair we shouldn't even include `callAsync`. It was only nice for testing purposes of unctx without using `withAsyncContext` in each test. In real use cases, `callAsync(obj)`, obj is an external dependency and `withAsyncContext` is a kinda special case but agree we could make them opt-in.